### PR TITLE
Add support for raising Python Exceptions from Nim Exceptions directly

### LINKED ISSUE #64: How to raise corresponding Python exception from Nim
If I have the following `function` that I expose into Python, 

```
proc function(path_to_file: string): string {. exportpy .} =
    try:
        some_nim_function(path_to_file)
    except:
        raise newException(Exception, "Unable to load file from " & path_to_file)
```

how do I get it to raise a `Exception` in Python if some_nim_function raises a `Exception` in Nim? Everything I've tried so far has resulted in Python interpreter dying.

### PR DESCRIPTION
**Pull request by kdheepak**
Fixes #64

### CODE REVIEW

File: tests/tnimfrompy.py [hunk 1]
```diff
@@ -2,6 +2,12 @@
 import nimfrompy as s
 
 assert(s.greet("world") == "Hello, world!")
+assert(s.greetEveryoneExceptJack("world") == "Hello, world!")
+try:
+    s.greetEveryoneExceptJack("Jack")
```
Comment (yglukhov) [inline tests/tnimfrompy.py]: `assert(false)` here should guarantee that the exception is in fact thrown.

File: nimpy.nim [hunk 2]
```diff
@@ -767,7 +770,13 @@ macro callNimProcWithPythonArgs(prc: typed, argsTuple: PPyObject): PPyObject =
         updateStackBottom()
         if verifyArgs(`argsTuple`, `argsLen`, `argsLenReq`, `nameLit`):
             `parseArgsStmts`
-            nimValueToPy(`origCall`)
+            try:
+                nimValueToPy(`origCall`)
+            except:
+                let e = getCurrentException()
+                let err = pyLib.PyErr_NewException("nimpy" & "." & $(e.name), pyLib.NimPyException, nil)
+                pyLib.PyErr_SetString(err, "Unexpected error encountered: " & getCurrentExceptionMsg())
```
Comment (yglukhov) [inline nimpy.nim]: Is `err` leaking?

Comment (kdheepak) [reply nimpy.nim]: I had a similar question, I believe so. `PyErr_NewException` creates a new Exception Python class object and increments the reference by 1, so I think it does result in a leak technically. Any suggestions on what do in this situation? Should we store the exception somewhere?

Comment (yglukhov) [reply nimpy.nim]: I think `decRef err` should work here.

Comment (kdheepak) [reply nimpy.nim]: I made the change and tests pass locally.

**Comment by kdheepak** [comment]
I don't know if this should be merged yet. On a larger codebase when I use this branch I seem to be getting a `SIGSEGV: Illegal storage access. (Attempt to read from nil?)` fatal error.

**Comment by yglukhov** [comment]
Oh ok. Is it `decRef err` that introduced it?

**Comment by kdheepak** [comment]
No I don't think so, I tested it before and after and still get the same segfault. I'll experiment more and post here.

**Comment by kdheepak** [comment]
I took the latest `master` and made just one change. I added a `try`, `except` to the `nimValueToPy` call

```

    result = quote do:
        updateStackBottom()
        if verifyArgs(`argsTuple`, `argsLen`, `argsLenReq`, `nameLit`):
            `parseArgsStmts`
            try:
                nimValueToPy(`origCall`)
            except:
                nimValueToPy(`origCall`)
        else:
            PPyObject(nil)

```

And this results in a segfault. Any idea what is going on here?

**Comment by yglukhov** [comment]
For the record. The crash was fixed in 701ffe15c00d8b11c1a571e682fb3297e2d3f0ee

**Comment by kdheepak** [comment]
Thank you for the quick fixes!

**Pull request merged**